inc/client_communication.hpp

```diff
@@ -173,24 +177,52 @@ class PackedClientEntry : public ClientEntryAbstract {
 
 };
 
+int8_t ParseMsg(uint8_t* rx_data, uint8_t rx_length,
+  ClientEntryAbstract** entry_array, uint8_t entry_length);
+
+int8_t ParseMsg(uint8_t* rx_data, uint8_t rx_length,
+  ClientEntryAbstract& entry);
+
 class ClientAbstract{
   public:
     ClientAbstract(uint8_t type_idn, uint8_t obj_idn):
       type_idn_(type_idn),
-      obj_idn_(obj_idn) {};
+      obj_idn_(obj_idn)
+      {};
 
     virtual ~ClientAbstract(){};
 
-    virtual void ReadMsg(uint8_t* rx_data, uint8_t rx_length) = 0;
+  int8_t ReadMsg(uint8_t* rx_data, uint8_t rx_length){
+    uint8_t number_of_entries = GetNumberOfClientEntries();
+    ClientEntryAbstract * client_array[number_of_entries];
+    GetClientEntryList(client_array);
 
-    const uint8_t type_idn_;
-    const uint8_t obj_idn_;
-};
+    int8_t parse_result = ParseMsg(rx_data, rx_length, client_array, number_of_entries);
 
-int8_t ParseMsg(uint8_t* rx_data, uint8_t rx_length,
-  ClientEntryAbstract** entry_array, uint8_t entry_length);
+    return parse_result;
+  }
 
-int8_t ParseMsg(uint8_t* rx_data, uint8_t rx_length,
-  ClientEntryAbstract& entry);
+    void UpdateModuleId(uint8_t new_id){
+      obj_idn_ = new_id;
```

Comment:
Should this call UpdateEntryIDs? Wondering if there's a specific reason to ever update  your module ID but not want to go and also update your entry IDs. May just be something worth automatically calling here whenever we update the module ID